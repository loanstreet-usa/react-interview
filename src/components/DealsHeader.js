import React, { Component } from 'react';
import PropTypes from 'prop-types';

class DealsHeader extends Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
        isSortField: PropTypes.bool.isRequired,
        onSort: PropTypes.func.isRequired,
    }
    render() {
        return (
            <th
                className="DealsTable--headerCell"
                onClick={this.invokeSort}
            >
                {this.sortIndicator()}
                {this.props.title}
            </th>
        );
    }
    invokeSort = () => {
        this.props.onSort(this.props.value);
    }
    sortIndicator() {
        return this.props.isSortField ? <span>*</span> : null;
    }
}

export default DealsHeader;