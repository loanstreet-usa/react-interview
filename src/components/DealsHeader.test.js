import React from 'react';
import { create} from 'react-test-renderer';
import DealsHeader from './DealsHeader';

it('Renders', () => {
    const mockHeader = {
        title: 'Mock Header',
        value: 'mock',
        isSortField: true,
        onSort: jest.fn(),
    };

    const target = create(<DealsHeader {...mockHeader} />)
    expect(target).toMatchSnapshot();
});
