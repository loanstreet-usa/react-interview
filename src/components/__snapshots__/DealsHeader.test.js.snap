// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<th
  className="DealsTable--headerCell"
  onClick={[Function]}
>
  <span>
    *
  </span>
  Mock Header
</th>
`;
